stagemonitor-core/src/main/java/org/stagemonitor/core/StagemonitorCoreConfigurationSourceInitializer.java

```diff
@@ -62,4 +77,71 @@ private void assertElasticsearchIsAvailable(ElasticsearchClient elasticsearchCli
 					"but elasticsearch is not reachable at " + corePlugin.getElasticsearchUrl(), e);
 		}
 	}
+
+	/**
+	 * Creates and registers a SpringCloudConfigConfigurationSource for each active profile, if the
+	 * SpringCloudConfigurationSource is enabled per flag
+	 */
+	private void addSpringCloudConfigurationSources(ConfigurationRegistry configuration, CorePlugin corePlugin) {
+		// Validating necessary properties
+		final String applicationName = corePlugin.getApplicationName();
+		final String springCloudConfigServerAddress = corePlugin.getSpringCloudConfigServerAddress();
+		if (StringUtils.isEmpty(applicationName) || CorePlugin.DEFAULT_APPLICATION_NAME.equals(applicationName)) {
+			logger.warn("stagemonitor.applicationName is not set (explicitly) but necessary for the config service configuration source." +
+					"Will skip the config source");
+			return;
+		}
+		if (StringUtils.isEmpty(springCloudConfigServerAddress) || !springCloudConfigServerAddress.startsWith("http")) {
```

Comment:
better validate in ConfigurationOption via ConfigurationOptionBuilder#addValidator

Reply:
done